### PR TITLE
Add Support for ONNX scripting Interpolate with missing shape

### PR DESCRIPTION
**Pull request by lara-hdr**
- Add support for missing case where interpolate is exported with missing shape information in scripting
- Add warnings

### CODE REVIEW

File: torch/onnx/symbolic_opset11.py [hunk 1]
```diff
@@ -85,17 +86,28 @@ def __interpolate(g, input, size, scale_factor, mode, align_corners):
...
+        input_size = sym_help._slice_helper(g, input_size, axes=[0], ends=[2], starts=[0])
+        # in some cases size is not a packed list but size is a scalar
+        # We need to also verify that (sym_help._maybe_get_const(size, 't').dim() == 0)
+        # but this information is not always available. Try to get the dim,
+        # and if not assume that it is not a scalar.
+        try:
+            is_scalar = not sym_help._is_packed_list(size) and ((sym_help._maybe_get_const(size, 't').dim() == 0))
+        except AttributeError:
+            warnings.warn("Cannot verify if the output_size is a scalar while exporting interpolate. Assuming that it is not a scalar.")
+            is_scalar = not sym_help._is_packed_list(size)
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py]: Should only need this warning if `is_scalar` is True.

File: torch/onnx/symbolic_opset11.py [hunk 2]
```diff
@@ -111,6 +123,8 @@ def __interpolate(g, input, size, scale_factor, mode, align_corners):
                     mode_s=mode,  # nearest, linear, or cubic
                     nearest_mode_s="floor")
     else:  # if not sym_help._is_none(scales)
+        if not input.isCompleteTensor():
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py]: Why is the complete tensor requirement a must have for the scales route?

File: torch/onnx/symbolic_opset11.py [hunk 3]
```diff
@@ -85,17 +86,28 @@ def __interpolate(g, input, size, scale_factor, mode, align_corners):
...
+        # but this information is not always available. Try to get the dim,
+        # and if not assume that it is not a scalar.
+        try:
+            is_scalar = not sym_help._is_packed_list(size) and ((sym_help._maybe_get_const(size, 't').dim() == 0))
+        except AttributeError:
+            warnings.warn("Cannot verify if the output_size is a scalar while exporting interpolate. Assuming that it is not a scalar.")
+            is_scalar = not sym_help._is_packed_list(size)
+
         if is_scalar:
+            if not input.isCompleteTensor():
```
Comment (BowenBao) [inline torch/onnx/symbolic_opset11.py]: nit: seems only rank info is needed, so this can be relaxed a bit.

**Review by BowenBao — CHANGES_REQUESTED** [review]

File: torch/onnx/symbolic_opset11.py (hunk 3 above)
Comment (lara-hdr) [reply torch/onnx/symbolic_opset11.py]: we access the dimension line 112. previously it was silently failing

File: torch/onnx/symbolic_opset11.py (hunk 2 above)
Comment (lara-hdr) [reply torch/onnx/symbolic_opset11.py]: input.type().dim() used line 128. just relaxed this to verify if rank is available.

**Comment by lara-hdr** [comment]
thank you for the comments @BowenBao, I addressed them in the last commit. thanks!

**Review by BowenBao — APPROVED** [review]

**Review by houseroad — APPROVED** [review]
LGTM

**Pull request closed without merging**